react-native/react/reducers/tracker.js

```diff
@@ -0,0 +1,206 @@
+'use strict'
+/* @flow */
+
+import * as Constants from '../constants/tracker'
+import {normal, warning, error, pending} from '../constants/tracker'
+
+import {identify} from '../keybase_v1'
+
+import type {UserInfo} from '../tracker/bio.render.desktop'
+import type {Proof} from '../tracker/proofs.render.desktop'
+import type {SimpleProofState} from '../constants/tracker'
+
+import type {Identity, RemoteProof, LinkCheckResult, ProofState} from '../constants/types/flow-types'
+import type {Action} from '../constants/types/flux'
+
+type State = {
+  serverStarted: boolean,
+  serverActive: boolean,
+  proofState: SimpleProofState,
+  username: ?string,
+  shouldFollow: ?boolean,
+  reason: string,
+  userInfo: UserInfo,
+  proofs: Array<Proof>
+}
+
+const initialProofState = pending
+
+const initialState: State = {
+  serverStarted: false,
+  serverActive: false,
+  username: null,
+  proofState: initialProofState,
+  shouldFollow: true,
+  proofs: [],
+  reason: 'TODO: get the reason',
+  userInfo: {
+    fullname: 'TODO: get this information',
+    followersCount: -1,
+    followingCount: -1,
+    followsYou: false,
+    avatar: 'TODO: get this information',
+    location: 'TODO: get this information'
+  }
+}
+
+export default function (state: State = initialState, action: Action): State {
+  switch (action.type) {
+    case Constants.onFollowChecked:
+      if (action.payload == null) {
+        return state
+      }
+      const shouldFollow: boolean = action.payload
+
+      return {
+        ...state,
+        shouldFollow
+      }
+    case Constants.updateUsername:
+      if (!action.payload) {
+        return state
+      }
+      const username = action.payload.username
+
+      return {
+        ...state,
+        username
+      }
+
+    case Constants.updateProofState:
+      const proofs = state.proofs
+      const allOk: boolean = proofs.reduce((acc, p) => acc && p.state === normal, true)
+      const anyWarnings: boolean = proofs.reduce((acc, p) => acc || p.state === warning, true)
+      const anyError: boolean = proofs.reduce((acc, p) => acc || p.state === error, false)
+      const anyPending: boolean = proofs.reduce((acc, p) => acc || p.state === pending, false)
+
+      let proofState: SimpleProofState = 'error'
+
+      if (allOk) {
+        proofState = 'normal'
```

Comment:
will fix, but fun fact if these aren't the correct constants it won't type check! :)